### DISCUSSION TITLE
Edit client framework on runtime

### DISCUSSION THREAD

**Discussion by CaptainKeys21** [serenity-rs/serenity#2362] [Q&A] [answered]
Hi! There is a way to get client framework and edit his properties while the bot is online? For example, change the bot prefix.

**Comment by arqunis** [comment, marked as answer]
Unfortunately, no. This would require revamping the code to allow this, which wouldn't be easy; however, if you specifically want this for changing prefixes, you should [take a look at `dynamic_prefix`](https://serenity-rs.github.io/serenity/current/serenity/framework/standard/struct.Configuration.html#method.dynamic_prefix). It grants you the ability to assign a prefix at runtime. You can use [`Context::data`](https://serenity-rs.github.io/serenity/current/serenity/client/struct.Context.html#structfield.data) to store prefixes you'd like your bot to use at a given moment, then retrieve those prefixes inside of `dynamic_prefix` and return the one you want.

**Reply by CaptainKeys21** [reply]
It was exactly what I was looking for, thanks!
